libraries/joomla/form/field.php

```
@@ -673,7 +681,12 @@ protected function getId($fieldId, $fieldName)
 	 *
 	 * @since   11.1
 	 */
-	abstract protected function getInput();
+	protected function getInput()
+	{
+		$layout = $this->getAttribute('layout', $this->layout);
```

we should throw an exception if the layout is null right?